src/global_env.cpp

```diff
@@ -44,19 +38,18 @@ Value GlobalEnv::global_get(Env *env, SymbolObject *name) {
         return NilObject::the();
 }
 
-Value GlobalEnv::global_set(Env *env, SymbolObject *name, Value val, bool check_readonly) {
+Value GlobalEnv::global_set(Env *env, SymbolObject *name, Value val, bool readonly) {
     std::lock_guard<std::recursive_mutex> lock(g_gc_recursive_mutex);
 
     if (!name->is_global_name())
         env->raise_name_error(name, "`{}' is not allowed as a global variable name", name->string());
 
     auto info = m_global_variables.get(name, env);
     if (info) {
-        if (check_readonly && info->is_readonly())
+        if (info->is_readonly())
             env->raise_name_error(name, "{} is a read only variable", name->string());
         info->set_object(val.object());
     } else {
-        auto readonly = readonly_globals.get(name->string()) != nullptr;
         auto info = new GlobalVariableInfo { val.object(), readonly };
         m_global_variables.put(name, info, env);
     }
```

Comment:
I think we should change line 49 to:
```c++
if (info->is_readonly() && !readonly)
```
Sometimes we have to reassign a global variable internally (e.g. `$?` after a subprocess has been run), that won't be possible with this code.

Reply:
Yeah good call.